src/devices/Nmea0183/SentenceCache.cs

```diff
@@ -92,35 +149,54 @@ public bool TryGetCurrentPosition(out GeographicPosition? position, out Angle tr
         /// GNSS receivers only deliver the position at 1Hz or less)
         /// </summary>
         /// <param name="position">Current position</param>
+        /// <param name="source">Only look at this source (otherwise, if multiple sources provide a position, any is used)</param>
         /// <param name="extrapolate">True to extrapolate the current position using speed and track</param>
         /// <param name="track">Track (course over ground)</param>
         /// <param name="sog">Speed over ground</param>
         /// <param name="heading">Vessel Heading</param>
+        /// <param name="messageTime">Time of the position report that was used</param>
+        /// <param name="now">The current time (when working with data in the past, this may be the a time within that data set)</param>
         /// <returns>True if a valid position is returned</returns>
-        public bool TryGetCurrentPosition(out GeographicPosition? position, bool extrapolate, out Angle track, out Speed sog, out Angle? heading)
+        public bool TryGetCurrentPosition(
```

Comment:
I might be missing something but why is this function in the SentenceCache? I'd expect this code to only do caching without any extra logic

Reply:
You are right, I separated these methods to a separate class. To much logic in the same class is poor design.